design/ceph-config-updates.md

```diff
@@ -0,0 +1,235 @@
+
+However, for it's own backwards compatibility, Ceph options set in the config file or via command
+line flags will override the centrally-configured settings. To make the most of this functionality
+within Ceph, it will be necessary to limit the configuration options specified by Rook in either
+config files or on the command line to a minimum.
+
+
+## Proposed new design
+
+The end goal of this work will be to allow Ceph to centrally manage its own configuration as much as
+possible. Or in other terms, Rook will specify the barest minimum of configuration options in the
+config file or on the command line with a priority on cleaning the config file.
+
+Rook currently offers the option of a config override, and this functionality can be kept for
+backwards compatibility for existing users; however, we should note that it is advisable not to do
+this and instead configure Ceph's centrally-managed options.
+
+### Minimal config file
+
+#### Required minimum
+
+All Ceph options in the config file can be set via the command line, so it is tempting to remove the
+need for having a `ceph.conf` in containers at all. In practice, we do not know how to make this an
+actual possibility.
+
+The troublesome situation regards Ceph daemons' ability to discover mons. The daemons must have
+`mon host` specified to find mons which are using non-default ports. It is further undesirable to
+specify this via the `--mon-host` flag because if this parameter is specified statically as a
+container argument and the initial mons fail and are replaced by new mons, the daemon will not know
+how to contact the mons if it is restarted.
+
+The proposed solution to this problem and the only config option which have thus far been deemed
+*required* is the `mon host` global option which will be stored in a Kubernetes ConfigMap so that
+containers always have the most updated version of this information. The single `ceph.conf`
+ConfigMap will be mounted to all containers at `/etc/ceph/ceph.conf`. There is a possibility the
+`mon initial members` global option may also be necessary, as it goes hand-in-hand with `mon host`.
+
+With this one -- or these two -- options set via the config file, all daemons should *require* no
+other config file options set.
+
+#### Convenient minimum
+
+There may be a small set of additional options which are okay to have in the config file for
+convenience if they would otherwise be set via command line for every single daemons.
+
+For example:
+- `fsid = <fsid>` is needed for mon fs bootstrapping, and it is set as a safety to make sure daemons
+  cannot connect to the wrong cluster accidentally
+- `log file = /dev/stderr` is set for all daemons to enforce the container standard of sending
+  logging to stdout/stderr
+- `mon cluster log file = /dev/stderr` for `log file` reasons above
+- `fatal signal handlers = false`
+  **TODO MORE INFORMATION NEEDED** why does Rook set this? It's purpose will determine its config location
```

Comment:
@travisn do you know what purpose this serves in Rook?

Reply:
I don't recall why this was added. @jbw976 do you recall?